wallet/src/de/schildbach/wallet/ui/dashpay/BottomNavFragment.kt

```diff
@@ -19,37 +19,33 @@ package de.schildbach.wallet.ui.dashpay
 
 import android.view.View
 import androidx.annotation.LayoutRes
-import androidx.fragment.app.Fragment
 import de.schildbach.wallet.ui.MainActivity
 import de.schildbach.wallet.ui.widget.KeyboardResponsiveCoordinatorLayout
 import kotlinx.android.synthetic.main.activity_main.*
+import org.dash.wallet.common.ui.BaseLockScreenFragment
 
-abstract class BottomNavFragment(@LayoutRes contentLayoutId: Int) : Fragment(contentLayoutId) {
+abstract class BottomNavFragment(@LayoutRes contentLayoutId: Int) : BaseLockScreenFragment(contentLayoutId) {
 
     var forceHideBottomNav: Boolean = false
 
     private val mainActivity by lazy {
-        requireActivity() as MainActivity
-    }
-
-    abstract val navigationItemId: Int
-    private val navigationItem by lazy {
-        mainActivity.bottom_navigation.menu.findItem(navigationItemId)
+        requireActivity() as? MainActivity
     }
 
     override fun onResume() {
         super.onResume()
         // select the right button in bottom nav
-        navigationItem.isChecked = true
         showHideBottomNav()
     }
 
     private fun showHideBottomNav() {
-        val navParentView = mainActivity.bottom_navigation.parent.parent
-        if (navParentView is KeyboardResponsiveCoordinatorLayout) {
-            navParentView.forceHideViewToHide = forceHideBottomNav
-        } else {
-            mainActivity.bottom_navigation.visibility = if (forceHideBottomNav) View.VISIBLE else View.GONE
+        mainActivity?.let { activity ->
+            val navParentView = activity.bottom_navigation.parent.parent
+            if (navParentView is KeyboardResponsiveCoordinatorLayout) {
+                navParentView.forceHideViewToHide = forceHideBottomNav
+            } else {
+                activity.bottom_navigation.visibility = if (forceHideBottomNav) View.VISIBLE else View.GONE
+            }
```

Comment:
`navigationItem.isChecked = true` logic is moved to `MainActivity`, but this one is left here for now. But I think it would be better to move it in there as well and get rid of `BottomNavFragment`. This will allow using the fragments that now are derived from it freely in other places, as well as help with the transition to the navigation component.